test/config_parser_SUITE.erl

```diff
@@ -266,7 +274,26 @@ hosts(_Config) ->
     ?err(parse(#{<<"general">> => #{<<"hosts">> => [<<>>]}})),
     ?err(parse(#{<<"general">> => #{<<"hosts">> => []}})),
     ?err(parse(#{<<"general">> => #{<<"hosts">> => [<<"host1">>, <<"host1">>]}})),
-    ?err(mongoose_config_parser_toml:parse(#{<<"general">> => #{}})). % hosts are mandatory
+    % either hosts or host_types must be provided
+    ?err(mongoose_config_parser_toml:parse(#{<<"general">> => #{}})).
+
+host_types(_Config) ->
+    %% ./rebar3 ct --suite config_parser_SUITE --group general --case host_types
+    ?eq([#config{key = host_types, value = [<<"type1">>, <<"type2">>]}],
+        parse(#{<<"general">> => #{<<"host_types">> => [<<"type1">>, <<"type2">>]}})),
+    ?eq([#config{key = host_types, value = [<<"what is this?">>]}],
+        parse(#{<<"general">> => #{<<"host_types">> => [<<"what is this?">>]}})),
```

Comment:
Maybe it's just me, but "what is this" looks like a wrong value, could you replace it with something like "my host type"?

Side note: I am not sure what would happen if we allow spaces here, they might be processed in the code as well-formed domain names, e.g. in places where process names are constructed from them.

Reply:
we should not have this restriction for host_types. 
host_types should be never processed as domain names. so whites paces seem to be a good option for sanity check.

"what is this"  it's leftover after copy past, will change it